torch/csrc/Generator.cpp

```diff
@@ -250,13 +250,60 @@ static PyObject* THPGenerator_get_device(THPGenerator* self, void* unused) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject* THPGenerator_reduce(PyObject* _self, PyObject* noargs) {
+  HANDLE_TH_ERRORS
+  auto self = (THPGenerator*)_self;
+  auto& gen = self->cdata;
+
+  auto ret = THPObjectPtr{PyTuple_New(3)};
+  if (!ret)
+    throw python_error();
+
+  py::object torch_module = py::module::import("torch");
+  py::object torch_generator = torch_module.attr("Generator");
+  PyTuple_SET_ITEM(ret.get(), 0, torch_generator.release().ptr());
```

Comment:
Instead of hardcoding the constructor, it would be better to have something like `_rebuild_from_type_v2`, which would make it easier to make breaking changes to serialization. In particular, the setup here has no way of evolving `__setstate__` in the future.